src/openms/include/OpenMS/FILTERING/DATAREDUCTION/MassTraceDetection.h

```diff
@@ -97,27 +103,93 @@ namespace OpenMS
 
         /** @name Private methods and members
         */
+
+       
     protected:
         void updateMembers_() override;
 
     private:
 
         struct Apex
         {
-          Apex(double intensity, Size scan_idx, Size peak_idx);
-          double intensity;
-          Size scan_idx;
-          Size peak_idx;
+          Apex(PeakMap& map, const Size scan_idx, const Size peak_idx);
+          // Default constructors
+          Apex() = default;
+          Apex(const Apex& other) = default;
+          Apex(Apex&& other) = default;
+
+          // Move assignment operator
+          Apex& operator=(Apex&& other) 
+          {
+            if (this != &other) {
+              map_ = other.map_;
```

Comment:
this is a copy of an MSExperiment. Ouch.
You really want to re-assign the reference here, but this cannot be done (std::move would move the MSExperiment from Other, which is also wrong).
What you need is use a reference_wrapper. See below